src/backend/marsha/core/tests/test_command_send_reminders.py

```diff
@@ -0,0 +1,995 @@
+                    out.getvalue(),
+                )
+                registration.refresh_from_db()
+                # key has been added
+                self.assertEqual(
+                    [REMINDER_3, REMINDER_2, REMINDER_1], registration.reminders
+                )
+                out.close()
+
+    def test_send_reminders_email_step_before_reminder_1(self):
+        """Reminders 3 and 2 haven't been sent, it's already time for  reminder 1,
+        make sure only reminder 1 is sent."""
+        video = VideoFactory(
+            live_state=IDLE,
+            live_type=RAW,
+            starting_at=timezone.now() + timedelta(minutes=4),
+        )
+
+        registration = LiveRegistrationFactory(
+            anonymous_id=uuid.uuid4(),
+            created_on=timezone.now() - timedelta(days=32),
+            is_registered=True,
+            should_send_reminders=True,
+            video=video,
+        )
+
+        out = StringIO()
+        call_command("send_reminders", stdout=out)
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertIn(
+            f"Sending email for liveregistration {registration.id} for video "
+            f"{registration.video.id} step {REMINDER_1}",
+            out.getvalue(),
+        )
+        registration.refresh_from_db()
+        # only key 1 has been added
+        self.assertEqual([REMINDER_1], registration.reminders)
+
+        # we call the command, no new email should be sent
+        call_command("send_reminders")
+        self.assertEqual(len(mail.outbox), 1)
+
+    def test_send_reminders_email_step_before_reminder_2(self):
+        """Reminders 3 haven't been sent, it's already time for  reminder 2,
+        make sure only reminder 2 is sent."""
+        video = VideoFactory(
+            live_state=IDLE,
+            live_type=RAW,
+            starting_at=timezone.now() + timedelta(hours=2),
+        )
+
+        registration = LiveRegistrationFactory(
+            anonymous_id=uuid.uuid4(),
+            created_on=timezone.now() - timedelta(days=32),
+            is_registered=True,
+            should_send_reminders=True,
+            video=video,
+        )
+
+        out = StringIO()
+        call_command("send_reminders", stdout=out)
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertIn(
+            f"Sending email for liveregistration {registration.id} for video "
+            f"{registration.video.id} step {REMINDER_2}",
+            out.getvalue(),
+        )
+        registration.refresh_from_db()
+        # only key 1 has been added
+        self.assertEqual([REMINDER_2], registration.reminders)
+
+        # we call the command, no new email should be sent
+        call_command("send_reminders")
+        self.assertEqual(len(mail.outbox), 1)
+
+    @patch(
+        "marsha.core.management.commands.send_reminders.send_mail",
+        side_effect=smtplib.SMTPException("Error SMTPException"),
+    )
```

Comment:
I think you should not patch at this level but only when the first command is called. Because for the second one you don't know it no mail are sent because the `send_mail` function has thrown an exception or because there is nothing to treat